src/reggae/reggae.d

```diff
@@ -404,25 +404,34 @@ private void writeIfDiffers(O)(auto ref O output, in string path, in string cont
 private string[] dubImportFlags(in imported!"reggae.options".Options options) {
     import reggae.dub.interop.dublib: fetchDubDeps, dubPackagePath;
     import std.file: exists;
+    import std.json: JSONType;
     import std.path: buildPath;
 
     fetchDubDeps(hiddenDirAbsPath(options));
 
-    const dubSourcePath = buildPath(dubPackagePath("dub", selectionsPkgVersion("dub")), "source");
+    // CTFE to check the JSON at compile-time
+    enum dubSelection = selectionsPkgVersion!"dub";
+    enum dubVersion = dubSelection.type == JSONType.string
+        ? dubSelection.str
+        : dubSelection["version"].str; // commit SHA
+
+    const dubSourcePath = buildPath(dubPackagePath("dub", dubVersion), "source");
```

Comment:
This *appears* to be working with a commit SHA from a git+https selection too. No idea whether that just happens to be working or is something we can rely on from the dub API side. @Geod24

Reply:
I'm on the fence about this, because having different providers stored in the same directory is icky. But I don't think I'll change it anytime soon.

Reply:
Oh, I've just seen that this whole `dubImportFlags` function is unused, and then that `dubPackagePath` function too...